File: java/src/main/java/io/cloudboost/api/CloudSearchApi.java
package io.cloudboost.api;

import com.sun.jersey.api.client.GenericType;

import io.cloudboost.ApiException;
import io.cloudboost.ApiClient;
import io.cloudboost.Configuration;
import io.cloudboost.Pair;

import io.cloudboost.model.CloudObject;
import io.cloudboost.model.CloudSearch;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-03-21T16:08:35.479Z")
public class CloudSearchApi {
  private ApiClient apiClient;

  public CloudSearchApi() {
    this(Configuration.getDefaultApiClient());
  }

  public CloudSearchApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * search for cloudobjects following the criteria defined herein
   * 
   * @param appId CloudApp id (required)
   * @param tableName cloud table name (required)
   * @param body query body (required)
   * @return List<CloudObject>
   * @throws ApiException if fails to make API call
   */
  public List<CloudObject> search(String appId, String tableName, CloudSearch body) throws ApiException {
    Object localVarPostBody = body;
    
    // verify the required parameter 'appId' is set
    if (appId == null) {
      throw new ApiException(400, "Missing the required parameter 'appId' when calling search");
    }
    
    // verify the required parameter 'tableName' is set
    if (tableName == null) {
      throw new ApiException(400, "Missing the required parameter 'tableName' when calling search");
    }
    
    // verify the required parameter 'body' is set
    if (body == null) {
      throw new ApiException(400, "Missing the required parameter 'body' when calling search");
    }
    
    // create path and map variables
    String localVarPath = "data/{app_id}/{table_name}/search".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "app_id" + "\\}", apiClient.escapeString(appId.toString()))
      .replaceAll("\\{" + "table_name" + "\\}", apiClient.escapeString(tableName.toString()));

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    GenericType<List<CloudObject>> localVarReturnType = new GenericType<List<CloudObject>>() {};
    return apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
    
  }
  
}

File: typescript-angular/API/Client/CloudSearchApi.ts
/// <reference path="api.d.ts" />

/* tslint:disable:no-unused-variable member-ordering */

namespace API.Client {
    'use strict';

    export class CloudSearchApi {
        protected basePath = 'http://api.cloudboost.io/';
        public defaultHeaders : any = {};

        static $inject: string[] = ['$http', '$httpParamSerializer'];

        constructor(protected $http: ng.IHttpService, protected $httpParamSerializer?: (d: any) => any, basePath?: string) {
            if (basePath) {
                this.basePath = basePath;
            }
        }

        private extendObj<T1,T2>(objA: T1, objB: T2) {
            for(let key in objB){
                if(objB.hasOwnProperty(key)){
                    objA[key] = objB[key];
                }
            }
            return <T1&T2>objA;
        }

        /**
         * search for cloudobjects following the criteria defined herein
         * 
         * @param appId CloudApp id
         * @param tableName cloud table name
         * @param body query body
         */
        public search (appId: string, tableName: string, body: CloudSearch, extraHttpRequestParams?: any ) : ng.IHttpPromise<Array<CloudObject>> {
            const localVarPath = this.basePath + 'data/{app_id}/{table_name}/search'
                .replace('{' + 'app_id' + '}', String(appId))
                .replace('{' + 'table_name' + '}', String(tableName));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling search');
            }
            // verify required parameter 'tableName' is set
            if (!tableName) {
                throw new Error('Missing required parameter tableName when calling search');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling search');
            }
            let httpRequestParams: any = {
                method: 'POST',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
    }
}
